oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/ClientSecretAuthenticationProvider.java

```diff
@@ -86,6 +90,7 @@ public Authentication authenticate(Authentication authentication) throws Authent
 
 		if (!ClientAuthenticationMethod.CLIENT_SECRET_BASIC.equals(clientAuthentication.getClientAuthenticationMethod()) &&
 				!ClientAuthenticationMethod.CLIENT_SECRET_POST.equals(clientAuthentication.getClientAuthenticationMethod())) {
+			logger.trace("Did not match client authentication method");
```

Comment:
Let's remove this - the log might get noisy. Apply the same to the other providers.